### PR TITLE
Fix Calo Point Time and add it PFP T0

### PR DESCRIPTION
**Pull request by gputnam**
Depends on https://github.com/SBNSoftware/sbnanaobj/pull/84.

### CODE REVIEW

**Comment by gputnam** [comment]
Added in another commit to make the CaloPoints usable.

File: sbncode/CAFMaker/FillReco.cxx
Comment (brucehoward-physics) [inline sbncode/CAFMaker/FillReco.cxx:573]: @grayputnam do we understand why the previous version wasn't looking as intended? I think it seems ok to use the hit `PeakTime()` but I guess I'm not sure *why* the previous version was not right.

(I guess this for loop is looping through all the hits to find the one which matches the calo point presumably because there is no art::Assn between calo point and hit? -- Just double checking I understand the purpose of this loop.)

Comment (gputnam) [reply sbncode/CAFMaker/FillReco.cxx:573]: Yes there is no association.

And I'm not sure why the x point method wasn't working... presumably Pandora is doing stuff with x position that I do not understand.

Comment (brucehoward-physics) [reply sbncode/CAFMaker/FillReco.cxx:573]: Pandora can change the X position for example if it stitches the track across the cathode, but I don't see why it would impact things in the way that Maria was seeing.

File: sbncode/CAFMaker/CAFMaker_module.cc
Comment (brucehoward-physics) [inline sbncode/CAFMaker/CAFMaker_module.cc:1683]: From the code conventions (and if I'm properly understanding what is returned here) I think this should be using a FindOneP or something that explicitly assumes there will only be one associated T0?

File: sbncode/CAFMaker/CAFMaker_module.cc
Comment (brucehoward-physics) [inline sbncode/CAFMaker/CAFMaker_module.cc:1732]: As commented above.

Comment (gputnam) [reply sbncode/CAFMaker/CAFMaker_module.cc]: Fixed.

File: sbncode/CAFMaker/CAFMaker_module.cc
Comment (gputnam) [reply sbncode/CAFMaker/CAFMaker_module.cc]: Fixed.

File: sbncode/CAFMaker/FillReco.cxx
Comment (gputnam) [reply sbncode/CAFMaker/FillReco.cxx:573]: I don't either. But the time variable was always meant to encode the hit time. I was trying to be cute and get it out of Pandora. Regardless of whether that "should" work, directly accessing the hit time is correct.

File: sbncode/CAFMaker/FillReco.cxx
Comment (brucehoward-physics) [inline sbncode/CAFMaker/FillReco.cxx:738]: Do we want to set some value for if !t0 case? If I'm not mistaken it will otherwise be nan in the CAF? Maybe I'm missing where some default is set.

File: sbncode/CAFMaker/CAFMaker_module.cc
Comment (brucehoward-physics) [inline sbncode/CAFMaker/CAFMaker_module.cc:1727]: I defer to others here because I'm definitely not proclaiming to be an expert in efficiency, but do we want to initialize this variable similarly to the pfpMeta below? (I.e. const and with a ternary.) Maybe the answer is no. Does @PetrilloAtWork have wisdom here?

File: sbncode/CAFMaker/CAFMaker_module.cc
Comment (brucehoward-physics) [inline sbncode/CAFMaker/CAFMaker_module.cc:1680]: Same as other instance of this.

**Review by brucehoward-physics — APPROVED** [review]
I include a possible suggestion and a question:

1) I defer to others here because I'm not proclaiming to be an expert in efficiency, but do we want to initialize this variable similarly to the pfpMeta below? (I.e. const and with a ternary.) Maybe the answer is no. Does @PetrilloAtWork have wisdom here?

2) If I'm reading correctly I think that the t0 variable is left nan in the case that t0 is NULL? Not sure if this is intended or if we want to set some default (or if I've misread). (For neutrino candidate slices it may not create any t0? Or for other things that don't stitch. And then we'd have lots of nans.)

However, not sure any of these should hold up me hitting approve so I'll hit approve. I assume that @grayputnam has done any necessary CI/validation? I have not done this.

**Pull request merged**
